### ISSUE TITLE
Bug in `lowPass(Field3D&, int zmax, int zmin)`?

### ISSUE THREAD

**Issue by ZedThree** [boutproject/BOUT-dev#1221]
In [`lowPass(const Field3D &var, int zmax, int zmin, REGION rgn)`:](https://github.com/boutproject/BOUT-dev/blob/master/src/field/field3d.cxx#L1126)

```cpp
    // Filter in z
    for(int jz=zmax+1;jz<=ncz/2;jz++)
      f[jz] = 0.0;

    // Filter zonal mode
    if(zmin==0) {
      f[0] = 0.0;
    }
```

`zmin` is an `int` but is only used if `zmin == 0`. Should it be a `bool`? Or is this supposed to do something different, like remove a particular mode?

**Comment by bendudson** [comment]
It's used in elm-pb as a bool:
https://github.com/boutproject/BOUT-dev/blob/master/examples/elm-pb/elm_pb.cxx#L1609
with `zmin == 1` meaning keep zonal (n=0) components and `zmin == 0` meaning take out zonal component. I agree this does not match what I'd expect for a name `zmin`. 

Perhaps replace `int zmin` with `bool keep_zonal`?

**Comment by ZedThree** [comment]
Fixed in #1591
